### PR TITLE
DAOS-623 ci: Add a simple pull request template

### PR DESCRIPTION
**Pull request by jolivier23**
This adds a PR submitter and reviewer template

Required-githooks: true

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      if githooks were not used.
+* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future
+      PRs.
+* [ ] All builds have passed.  Check non-required builds to ensure they don't indicate any problem
+      such as a compiler warning on extraneous platforms.
+* [ ] No new NLT or valgrind warnings.  Check the classic view. This step should only matter if
+      the build requires force landing.
+* [ ] Ensure sufficent testing is done.   Check feature pragmas and test tags.
+* [ ] Quick-build or Quick-functional is not used.
+* [ ] Pay attention to PRs that may effect compatibility between versions and ensure it has been
```
Comment (kccain) [inline .github/pull_request_template.md:24]: very minor suggestion: effect -> affect

**Review by kccain — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+### Before requesting gatekeeper:
+
+* [ ] Two review approvals and any prior change requests have been resolved.
+* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it
+      should be force landed and forced-landing tag is set
+
```
Comment (phender) [inline .github/pull_request_template.md:6]: Suggestions to add:

- the PR owner should ensure the commit message follows the standard before requesting gatekeeper
- the PR owner should also ensure any tests skipped by the ticket being addressed by the PR have been run and passed in the PR

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      changes.
+* [ ] Githooks were used. If not, there is a sufficient reason to move forward.  Check copyrights
+      if githooks were not used.
+* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future
+      PRs.
+* [ ] All builds have passed.  Check non-required builds to ensure they don't indicate any problem
+      such as a compiler warning on extraneous platforms.
+* [ ] No new NLT or valgrind warnings.  Check the classic view. This step should only matter if
+      the build requires force landing.
+* [ ] Ensure sufficent testing is done.   Check feature pragmas and test tags.
```
Comment (phender) [inline .github/pull_request_template.md:22]: The gatekeeper should also verify that any tests skipped by the ticket sourcing the PR now pass with the PR changes

**Review by phender — CHANGES_REQUESTED** [review]

**Review by kccain — APPROVED** [review]

**Review by phender — APPROVED** [review]

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+### Before requesting gatekeeper:
+
+* [ ] Two review approvals and any prior change requests have been resolved.
```
Comment (ashleypittman) [inline .github/pull_request_template.md:3]: There's a GitHub actions hook for "review-requested" so we could check for two +1 approvals when gatekeeper is added and add a comment to the PR saying "please don't do that" which would have the same effect but require zero additional work from gatekeepers.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it
+      should be force landed and forced-landing tag is set.
```
Comment (ashleypittman) [inline .github/pull_request_template.md:5]: "PR is green or force-landing label is set".  Again, we could check on this using the above trigger.

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+### Gatekeeper:
+
+* [ ] You are the appropriate gatekeeper to be landing the patch.
+* [ ] The PR has 2 reviews by people familiar with the code.
+* [ ] Any appropriate watchers have had a chance to review the PR.
+* [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested
+      changes.
+* [ ] Githooks were used. If not, there is a sufficient reason to move forward.  Check copyrights
+      if githooks were not used.
```
Comment (ashleypittman) [inline .github/pull_request_template.md:18]: We really should add a proper CI check for copyright dates, and we should do it before the next change-of-year.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future
+      PRs.
+* [ ] All builds have passed.  Check non-required builds to ensure they don't indicate any problem
+      such as a compiler warning on extraneous platforms.
```
Comment (ashleypittman) [inline .github/pull_request_template.md:22]: This is step 2, PR is Green isn't it?  All compiler steps should be marked as required except the landings-builds ones which only trigger if you modify certain files.

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+* [ ] No new NLT or valgrind warnings.  Check the classic view. This step should only matter if
+      the build requires force landing.
```
Comment (ashleypittman) [inline .github/pull_request_template.md:24]: Only applies for force-landing.

**Review by ashleypittman — COMMENTED** [review]
Interesting idea but there are a lot of steps here which is a big burden on gatekeepers and some of them would be better enforced via other means IMO.

**Pull request merged**